src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp

```diff
@@ -119,7 +121,12 @@ void GcodeSuite::M29() {
 
 // M30 - Delete a file on the SD card
 void GcodeSuite::M30() {
-    // TODO
+    ArrayStringBuilder<FILE_PATH_MAX_LEN> filepath;
+    filepath.append_printf("/usb/%s", parser.string_arg);
+    DeleteResult result = remove_file(filepath.str());
```

Comment:
We should check `builder.is_ok()` and prevent deletion attempt if the string was cropped

Reply:
ok, checking via is_ok as well